kubechain/internal/controller/taskruntoolcall/taskruntoolcall_controller.go

```diff
@@ -541,6 +578,31 @@ func (r *TaskRunToolCallReconciler) requestHumanApproval(ctx context.Context, tr
 		return ctrl.Result{}, nil
 	}
 
+	// Verify HLClient is initialized
+	if r.HLClientFactory == nil {
+		err := fmt.Errorf("HLClient not initialized")
+		result, errStatus, _ := r.setStatusError(ctx, kubechainv1alpha1.TaskRunToolCallPhaseErrorRequestingHumanApproval,
+			"NoHumanLayerClient", trtc, err)
+		return result, errStatus
+	}
+
+	// Post to HumanLayer to request approval
+	functionCall, statusCode, err := r.postToHumanLayer(ctx, trtc, contactChannel, apiKey)
+	if err != nil {
+		errorMsg := fmt.Errorf("HumanLayer request failed with status code %d: %v", statusCode, err)
+		result, errStatus, _ := r.setStatusError(ctx, kubechainv1alpha1.TaskRunToolCallPhaseErrorRequestingHumanApproval,
+			"HumanLayerRequestFailed", trtc, errorMsg)
+		return result, errStatus
+	}
+
+	// Update with call ID and requeue
+	callId := functionCall.GetCallId()
+	trtc.Status.ExternalCallID = callId
+	if err := r.Status().Update(ctx, trtc); err != nil {
```

Comment:
this is not following the pattern of doing a deepcopy for status changes. Maybe that's ok?